test/tests/mortar/periodic_segmental_constraint/testperiodicbase.i

```diff
@@ -0,0 +1,177 @@
+    new_boundary = '10 11 12 13'
+  []
+  [left_block_id]
+    type = SubdomainIDGenerator
+    input = left_block_sidesets
+    subdomain_id = 1
+  []
+  [left]
+    type = LowerDBlockFromSidesetGenerator
+    input = left_block_id
+    sidesets = '13'
+    new_block_id = '10003'
+    new_block_name = 'secondary_left'
+  []
+  [right]
+    type = LowerDBlockFromSidesetGenerator
+    input = left
+    sidesets = '11'
+    new_block_id = '10001'
+    new_block_name = 'primary_right'
+  []
+  [bottom]
+    type = LowerDBlockFromSidesetGenerator
+    input = right
+    sidesets = '10'
+    new_block_id = '10000'
+    new_block_name = 'secondary_bottom'
+  []
+  [top]
+    type = LowerDBlockFromSidesetGenerator
+    input = bottom
+    sidesets = '12'
+    new_block_id = '10002'
+    new_block_name = 'primary_top'
+  []
+
+  [corner_node]
+    type = ExtraNodesetGenerator
+    new_boundary = 'pinned_node'
+    nodes = '0'
+    input = top
+  []
+[]
+
+[Variables]
+  [u]
+    order = FIRST
+    family = LAGRANGE
+  []
+  [kappa]
+    order = SECOND
+    family = SCALAR
+  []
+[]
+
+[AuxVariables]
+  [kappa_aux]
+    order = SECOND
+    family = SCALAR
+  []
+[]
+
+[AuxScalarKernels]
+  [kappa]
+    type = FunctionScalarAux
+    variable = kappa_aux
+    function = '1 3'
+    execute_on = initial #timestep_end
+  []
+[]
+
+[Kernels]
+  [diff1]
+    type = Diffusion
+    variable = u
+    block = 1
+  []
+[]
+
+[Problem]
+  kernel_coverage_check = false
+  error_on_jacobian_nonzero_reallocation = true
+[]
+
+[BCs]
+  [fix_right]
+    type = DirichletBC
+    variable = u
+    boundary = pinned_node
+    value = 0
+  []
+[]
+
+[Constraints]
+  [mortarlr]
+    type = PenaltyEqualValueConstraint
+    primary_boundary = '11'
+    secondary_boundary = '13'
+    primary_subdomain = 'primary_right'
+    secondary_subdomain = 'secondary_left'
+    secondary_variable = u
+    correct_edge_dropping = true
+    penalty_value = 1.e3
+  []
+  [periodiclr]
+    type = PenaltyPeriodicSegmentalConstraint
+    primary_boundary = '11'
+    secondary_boundary = '13'
+    primary_subdomain = 'primary_right'
+    secondary_subdomain = 'secondary_left'
+    secondary_variable = u
+    kappa = kappa
+    coupled_scalar = kappa
+    kappa_aux = kappa_aux
+    correct_edge_dropping = true
+    pen_scale = 1.e3
+  []
+  [mortarbt]
+    type = PenaltyEqualValueConstraint
+    primary_boundary = '12'
+    secondary_boundary = '10'
+    primary_subdomain = 'primary_top'
+    secondary_subdomain = 'secondary_bottom'
+    secondary_variable = u
+    correct_edge_dropping = true
+    penalty_value = 1.e3
+  []
+  [periodicbt]
+    type = PenaltyPeriodicSegmentalConstraint
+    primary_boundary = '12'
+    secondary_boundary = '10'
+    primary_subdomain = 'primary_top'
+    secondary_subdomain = 'secondary_bottom'
+    secondary_variable = u
+    kappa = kappa
+    coupled_scalar = kappa
+    kappa_aux = kappa_aux
+    correct_edge_dropping = true
+    pen_scale = 1.e3
```

Comment:
It looks like this file is not actually tested, otherwise we would have gotten a warning (turned into an error) about the unused parameter. So we should either add a test for this or just remove it

Reply:
I'm removing this test file, as the `penalty_periodic_simple2d.i` file does basically the same thing, although the macro/average flux is different.